src/main/webapp/header.jsp

```diff
@@ -275,13 +275,18 @@ finally{
             function startCountdown() {
 
                 var warningCountdownTime = <%= sessionCountdownTime %>; // Session countdown time in minutes.
-                var countdownTime = warningCountdownTime * 60;
+                var countdownTimeout = warningCountdownTime * 60 * 1000;
 
                 countdownInterval = setInterval(function()
                 {
-                    updateCountdownDisplay(countdownTime);
-                    countdownTime = countdownTime - 1;
-                    if (countdownTime <= 0) {
+
+                    var now = Date.now();
+                    lastActivityTimestamp = parseInt(localStorage.getItem('lastActivity'));
```

Comment:
i wonder if there ever is the case where localStorage `lastActivity` is not set? in this case, `parseInt(...)` would get a `NaN` i believe. if that is a worry, you might add something like ` || 0` to the end to have a fallback?  just an idea...

Reply:
`resetActivity()` sets the value of `lastActivity`  and its  the first function to execute so its value will be always set before this function executes

Reply:
sounds good then. :+1: